istioctl/cmd/analyze.go

```diff
@@ -115,15 +115,35 @@ istioctl experimental analyze -k -d false
 				}
 			}
 
-			messages, err := sa.Analyze(cancel)
+			result, err := sa.Analyze(cancel)
 			if err != nil {
 				return err
 			}
 
-			for _, m := range messages {
-				fmt.Fprintf(cmd.OutOrStdout(), "%v\n", m.String())
+			// Maybe output details about which analyzers ran
+			if verbose {
+				if len(result.SkippedAnalyzers) > 0 {
+					fmt.Fprintln(cmd.OutOrStdout(), "Skipped analyzers:")
+					for _, a := range result.SkippedAnalyzers {
+						fmt.Fprintln(cmd.OutOrStdout(), "\t", a)
+					}
+				}
+				if len(result.ExecutedAnalyzers) > 0 {
+					fmt.Fprintln(cmd.OutOrStdout(), "Executed analyzers:")
+					for _, a := range result.ExecutedAnalyzers {
+						fmt.Fprintln(cmd.OutOrStdout(), "\t", a)
+					}
+				}
+				fmt.Fprintln(cmd.OutOrStdout())
 			}
 
+			if len(result.Messages) == 0 {
+				fmt.Fprintln(cmd.OutOrStdout(), "\u2714 No validation issues found.")
```

Comment:
If we were being really pedantic we should probably check for a unicode enabled terminal before we do anything fancy. On Linux at least I think you can check $LANG for a UTF-8 substring.

> $> echo $LANG
> en_US.UTF-8

Reply:
Also consider stderr.

Reply:
Switching this (and the other verbose output) to stderr makes sense.

Reply:
Actually isatty is a bit different. That just checks to see whether the output is going to a terminal or not, the issue here is that there may be something on the other end but it might not support unicode.

Consider case one, where I'm piping output to another bash command. It can support unicode, but can't read color. Case two, I'm running on a 2000s era Solaris box via telnet and run istioctl analyze. I am a tty, but can't support unicode.

But I'm 100% fine leaving this out of this diff. I imagine there are very, very few people who are going to run istioctl analyze that don't support unicode. It might make a good onboarding task  for someone if we open an issue and let it sit.